pkg/artifactory/resource_xray_watch_test.go

```diff
@@ -0,0 +1,329 @@
+package artifactory
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/xero-oss/go-xray/xray"
+)
+
+func TestAccWatch_basic(t *testing.T) {
+	watchName := "test-watch"
+	policyName := "test-policy"
+	watchDesc := "watch created by xray acceptance tests"
+	resourceName := "xray_watch.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		CheckDestroy: testAccCheckWatchDestroy,
+		Providers:    testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccXrayWatch_basic(watchName, watchDesc, policyName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", watchName),
+					resource.TestCheckResourceAttr(resourceName, "description", watchDesc),
+					resource.TestCheckResourceAttr(resourceName, "resources.0.type", "all-repos"),
+					resource.TestCheckResourceAttr(resourceName, "assigned_policies.0.name", policyName),
+					resource.TestCheckResourceAttr(resourceName, "assigned_policies.0.type", "security"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: false,
+			},
+			{
+				Config: testAccXrayWatch_unassigned(policyName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWatchDoesntExist(resourceName),
+				),
+			},
+		},
+	})
+}
+
+// These two tests are commented out because repoName and binMgrId must be real values but neither are terraformable so can't be put into these tests
```

Comment:
Please explain

Reply:
The API requires that `binMgrId` and `repoName` refer to real artifactory instances and repositories respectively. There's no way currently to create either of those resources with terraform, so I can't for example have a `resource "artifactory_bin_mgr" "foo"` and then `binMgrId := artifactory_bin_mgr.foo.id`. I tested this manually using the values from an Xray test instance I had access to, but it didn't make sense to leave those specific values in the code since nobody else would be able to use them; however, I left the tests in as it would be useful if they were to be run against another test Xray instance. The commented out tests can be ignored or deleted as you prefer.